R/helpers-pattern.R

```diff
@@ -144,7 +146,23 @@ matches <- function(match,
                     vars = NULL) {
   check_match(match)
   vars <- vars %||% peek_vars(fn = "matches")
-  flat_map_int(match, grep_vars, vars, ignore.case = ignore.case, perl = perl)
+
+  if (inherits(match, "pattern") || inherits(match, "stringr_pattern")) {
+    check_installed("stringr")
+    if (!missing(ignore.case)) {
+      cli::cli_abort("{.arg ignore.case} not supported when {.arg match} is a stringr pattern")
+    }
+    if (!missing(perl)) {
+      cli::cli_abort("{.arg perl} not supported when {.arg match} is a stringr pattern")
+    }
+    if (length(match) > 1) {
+      cli::cli_abort("stringr patterns must be length 1")
```

Comment:
```suggestion
      cli::cli_abort("stringr patterns must be length 1.")
```

Reply:
Could you use `flat_map_int()` as with `grep_vars` here?

Reply:
No, because stringr doesn't provide a `[` or `[[` method so all the attributes are lost 😬

Reply:
Ah got it. It would make sense to mention this in a comment.

Not important but we forgot about using the `{.pkg }` style here.

Reply:
I'll just commit those changes direct to main.